pkg/querier/batch/merge.go

```diff
@@ -112,8 +112,7 @@ func (c *mergeIterator) buildNextBatch(size int) bool {
 	for len(c.h) > 0 && (len(c.batches) == 0 || c.nextBatchEndTime() >= c.h[0].AtTime()) {
 		c.nextBatchBuf[0] = c.h[0].Batch()
 		c.batchesBuf = mergeStreams(c.batches, c.nextBatchBuf[:], c.batchesBuf, size)
-		copy(c.batches[:len(c.batchesBuf)], c.batchesBuf)
-		c.batches = c.batches[:len(c.batchesBuf)]
```

Comment:
This is a no-op, right?  Did it impact performance?

Reply:
My guess about this change (but Bryan can confirm or negate), is that we had to do this change because `c.batches` may need to grow after the change in `newMergeIterator()`. @bboreham is my understanding correct?

Reply:
Yes the `append` will grow the slice if required, whereas `copy` will panic.  `TestMergeIter/DoubleDelta` fails if you don't make this change.